errors/list.go

```diff
@@ -84,11 +92,30 @@ func (l *List) Detailed() string {
 
 // Append appends a new error on the error list.
 // If the error is nil it will not be added on the error list.
+//
+// Any error of type hcl.Diagnostics will be flattened inside
+// the error list, each hcl.Diagnostic will become an error.Error.
+//
+// Any error of type errors.List will be flattened inside
+// the error list.
 func (l *List) Append(err error) {
 	if err == nil {
 		return
 	}
-	l.errs = append(l.errs, err)
+	switch e := err.(type) {
+	case hcl.Diagnostics:
```

Comment:
:thinking: 
strange this also here.
Why not use like `errs.Append(errors.E(diags))` ?